scalableminds/webknossos
# Improve google analytics usage

This PR adds a generic click logger for google analytics. Every click on an element with a `textContent` that is not empty will be send to ga. This allows us to track the usage of features such as the NML download, the abstract tree tab, dataset settings and many more.
During development I noticed that although we already track all redux actions, most of them register for a wrong page (for example, the dashboard instead of the tracing view). This PR fixes that.
I also noticed that loading the tracing view always triggered a complete fresh page load (including .js/.css/.html download, html parsing etc.), due to the usage of `location.href = ...` to load the page. This PR fixes that as well by using the react router `history.push` functionality. This should result in a slightly better page load performance :)

We'll evaluate how well the click tracking works in a week or so and refine it afterwards.

/cc: @hotzenklotz @normanrz 

### URL of deployed dev instance (used for testing):
- https://gaclicks.webknossos.xyz

### Steps to test:
- Click some buttons and have a look at the network tab to see the click events that are sent to ga (disable your adblocker for that). The url that is sent in the requests should be correct.
- Open a new skeleton tracing and monitor the network tab, the page should not be reloaded, there should be no html/js/css loading.

### Issues:
- fixes #1293 

------
